osu.Game/Overlays/Toolbar.cs

```diff
@@ -87,26 +121,36 @@ public override void Load(BaseGame game)
                     Direction = FlowDirection.HorizontalOnly,
                     RelativeSizeAxes = Axes.Y,
                     AutoSizeAxes = Axes.X,
-                    Children = new []
+                    Children = new Drawable[]
                     {
                         new ToolbarButton
                         {
                             Icon = FontAwesome.search
                         },
-                        new ToolbarButton
-                        {
-                            Icon = FontAwesome.user,
-                            Text = ((OsuGame)game).Config.Get<string>(OsuConfig.Username)
-                        },
+                        toolbarUserButton = new UserButton(1),
                         new ToolbarButton
                         {
                             Icon = FontAwesome.bars
                         },
                     }
                 }
             };
+
+            InitUser();
+            Scheduler.AddDelayed(delegate {
+                InitUser();
+            }, 10000, true);
+
+            RelativeSizeAxes = Axes.X;
+            Size = new Vector2(1, height);
         }
 
+        protected override void Update()
+        {
+            base.Update();
+            api.Update();

```

Comment:
this is not intended use. you're taking Game's api and then updating it here a second time.